libraries/api-documenter/src/Documenter.ts

```diff
@@ -0,0 +1,248 @@
+              [Domifier.createCode(RenderingHelpers.getConciseSignature(memberName, member), 'javascript')],
+              RenderingHelpers.getDocId(apiJsonFile.packageName, className, memberName))
+          ];
+
+          methodsTable.rows.push(
+            Domifier.createTableRow([
+              methodTitle,
+              member.accessModifier ? [Domifier.createCode(member.accessModifier.toString(), 'javascript')] : [],
+              member.returnValue ? [Domifier.createCode(member.returnValue.type, 'javascript')] : [],
+              Domifier.renderDocElements(member.summary)
+            ])
+          );
+          this._writeMethodPage(member, memberName, className, apiJsonFile, renderer);
+          break;
+      }
+    }
+
+    if (methodsTable.rows.length > 0) {
+      domPage.elements.push(Domifier.createHeading1('Methods'));
+      domPage.elements.push(methodsTable);
+    }
+
+    if (docClass.remarks && docClass.remarks.length) {
+      domPage.elements.push(Domifier.createHeading1('Remarks'));
+      domPage.elements.push(...Domifier.renderDocElements(docClass.remarks));
+    }
+
+    renderer.writePage(domPage);
+  }
+
+  private _writeMethodPage(docMethod: IDocMethod, methodName: string, className: string,
+    apiJsonFile: ApiJsonFile, renderer: BasePageRenderer): void {
+
+    const fullMethodName: string = className + '.' + methodName;
+
+    const domPage: IDomPage = Domifier.createPage(`${fullMethodName} method`,
+      RenderingHelpers.getDocId(apiJsonFile.packageName, className, methodName));
+
+    if (docMethod.isBeta) {
+      this._writeBetaWarning(domPage.elements);
+    }
+
+    domPage.elements.push(...Domifier.renderDocElements(docMethod.summary));
+
+    domPage.elements.push(Domifier.PARAGRAPH);
+    domPage.elements.push(...Domifier.createTextElements('Signature:', { bold: true }));
+    domPage.elements.push(Domifier.createCodeBox(docMethod.signature, 'javascript'));
+
+    if (docMethod.returnValue) {
+      domPage.elements.push(...Domifier.createTextElements('Returns:', { bold: true }));
+      domPage.elements.push(...Domifier.createTextElements(' '));
+      domPage.elements.push(Domifier.createCode(docMethod.returnValue.type, 'javascript'));
+      domPage.elements.push(Domifier.PARAGRAPH);
+      domPage.elements.push(...Domifier.renderDocElements(docMethod.returnValue.description));
+    }
+
+    if (docMethod.remarks && docMethod.remarks.length) {
+      domPage.elements.push(Domifier.createHeading1('Remarks'));
+      domPage.elements.push(...Domifier.renderDocElements(docMethod.remarks));
+    }
+
+    const parametersTable: IDomTable = Domifier.createTable([
+      Domifier.createTextElements('Parameter'),
+      Domifier.createTextElements('Type'),
+      Domifier.createTextElements('Description')
+    ]);
+
+    domPage.elements.push(Domifier.createHeading1('Parameters'));
+    domPage.elements.push(parametersTable);
+    for (const parameterName of Object.keys(docMethod.parameters)) {
```

Comment:
a function might not have parameters

Reply:
Good point

---
In reply to: [135167892](https://github.com/Microsoft/web-build-tools/pull/322#discussion_r135167892) [](ancestors = 135167892)